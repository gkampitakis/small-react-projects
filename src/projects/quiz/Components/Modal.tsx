import React, { ReactElement } from 'react';
import { useGlobalContext } from '../context';

export default function Modal (): ReactElement {
  const { closeModal, isModalOpen, correct, questions } = useGlobalContext();

  return (
    <div
      className={`${isModalOpen ? 'modal-container isOpen' : 'modal-container'
        }`}
    >
      <div className='modal-content'>
        <h2>congrats!</h2>
        <p>
          You answered {((correct / questions.length) * 100).toFixed(0)}% of
        questions correctly
      </p>
        <button className='close-btn' onClick={closeModal}>
          play again
      </button>
      </div>
    </div>
  );
}
