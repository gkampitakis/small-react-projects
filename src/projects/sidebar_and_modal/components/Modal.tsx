import React from 'react';
import { useGlobalContext } from '../context';
import { FaTimes } from 'react-icons/fa';

export default function Modal () {
  const { closeModal, isModalOpen } = useGlobalContext();


  return (
    <div className={isModalOpen ? "modal-overlay show-modal" : "modal-overlay"}>
      <div className="modal-container">
        <h3>Modal Content</h3>
        <button className="close-modal-btn" onClick={closeModal}>
          <FaTimes />
        </button>
      </div>
    </div>
  )
}
