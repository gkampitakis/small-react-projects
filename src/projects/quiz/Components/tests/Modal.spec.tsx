import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Modal from '../Modal';

jest.mock('../../context', () => {
  let isOpen = true;

  return {
    useGlobalContext: () => ({
      closeModal: () => isOpen = !isOpen,
      isModalOpen: isOpen,
      correct: 2,
      questions: ['', '', '']
    })
  };
});

describe('Modal', () => {
  const { useGlobalContext } = jest.requireMock('../../context');

  it('Should render correctly', () => {
    const tree = renderer
      .create(<Modal />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should close modal on button click', () => {
    render(<Modal />);

    userEvent.click(screen.getByText(/play again/i));

    expect(useGlobalContext().isModalOpen).toBe(false);
  });
});
