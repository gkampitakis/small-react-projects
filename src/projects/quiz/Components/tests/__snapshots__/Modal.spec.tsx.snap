// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Should render correctly 1`] = `
<div
  className="modal-container isOpen"
>
  <div
    className="modal-content"
  >
    <h2>
      congrats!
    </h2>
    <p>
      You answered 
      67
      % of questions correctly
    </p>
    <button
      className="close-btn"
      onClick={[Function]}
    >
      play again
    </button>
  </div>
</div>
`;
